lib/Property/ICalendar/DateTime.php

```diff
@@ -343,8 +343,8 @@ public function validate($options = 0)
         $messages = parent::validate($options);
         $valueType = $this->getValueType();
         $values = $this->getParts();
-        try {
-            foreach ($values as $value) {
+        foreach ($values as $value) {
```

Comment:
which error is fixed here? it looks to me like the logic did change (before we stopped iteration on the first exception.. now iterate all values, no matter a exception is thrown in between)

Reply:
`$value` is used in the catch, which wouldn't be set if `$values` were empty.

This would indeed change the logic a bit; the user would receive all error messages instead of only the first one. I see that as a minor improvement.

Alternatively, `$value` could be set to null before the foreach, as that would also fix the phpstan error.

Reply:
feel free to suggest logic changes as separate PRs and concentrate on fixing just phpstan issues please.

otherwise we might introduce bugs

Reply:
Ok. I will revert this and set `$value` to null before the foreach.

Reply:
@JeroenVanOort I don't see where you have reverted this logic change and done the `$value = null` or `$value = ''` thing?
What is the status?

Reply:
Soon after I made that comment, we discussed that setting a loop variable outside the loop should be avoided. That's why I kept the change, but added the break statement in order to prevent logic changes.

Reply:
Thanks, I didn't notice the bonus `break;` at line 357!